libs/langchain/langchain/vectorstores/pgvector.py

```diff
@@ -301,6 +301,46 @@ def add_texts(
             texts=texts, embeddings=embeddings, metadatas=metadatas, ids=ids, **kwargs
         )
 
+    def update_documents(
```

Comment:
@lorenzofavaro take a look at the indexing code that @hwchase17 referenced. It should be able to solve this use case.
I'm OK adding an `upsert_documents` functionality, but it would need to be added on the base class as well, and would require the user to provide `ids` as part of the interface, and would need to implement upsert semantics.

---

The code should not assume that the content is identical just from the page_content, as there are use-cases when the relevant content lives in the metadata (e.g., metadata about two different products that share the same basic description). In this case, we'd want both documents to be indexed.

Reply:
@eyurtsev I worked on this change, I implemented the upsert semantics in the last commit.

Actually if the `upsert_documents` is also included in the base class (`VectorStore`) it must also be implemented in all other vector stores (apart from `PGVector`). 

If it's okay I can start adding this functionality for `PGVector`, adding the abstract method `upsert_documents` in the base class (in another commit) and in other PRs I can work on the others vector stores.